includes/prometheus.inc.php

```diff
@@ -26,19 +26,28 @@ function prometheus_push($device, $measurement, $tags, $fields)
 
                 foreach ($fields as $k => $v) {
                     if ($v !== null) {
+                        $k = preg_replace('/([a-z]{1,})/', '$1_', $k); // Generally, metric naming in Prometheus is done with underscores without capital letters
+                        $k = substr($k, 0, -1);
+                        $k = preg_replace('/([_]{2})/', '_', $k);
+                        $k = strtolower($k);
                         $vals = $vals . "$k $v\n";
                     }
                 }
-                
+
                 foreach ($tags as $t => $v) {
                     if ($v !== null) {
+                        if (strpos($v, "/") === false) {
```

Comment:
Does this not make more sense to just run str_replace() on $v all the time?

Reply:
Sure, did that. A haven't thought about a filesystem that consists of "/" itself, that is, a root filesystem. So it is needed to substitute with something. For now I added "_" symbol as a substitution. There are no associated errors produced.